src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.SingleCheckout/ReviewCartSingleCheckoutController.ts

```diff
@@ -45,41 +49,38 @@ module Orckestra.Composer {
 
                         if (!this.debounceUpdateItem) {
                             this.debounceUpdateItem = _.debounce(id => {
-                                this.Mode.Loading = true;
                                 let itemToUpdate = _.find(this.Cart.LineItemDetailViewModels, (i: any) => i.Id === id);
                                 self.checkoutService.updateCartItem(itemToUpdate.Id,
                                     itemToUpdate.Quantity,
                                     itemToUpdate.ProductId,
-                                    itemToUpdate.RecurringOrderFrequencyName === '' ? null : itemToUpdate.RecurringOrderFrequencyName,
+                                    itemToUpdate.RecurringOrderFrequencyName ? itemToUpdate.RecurringOrderFrequencyName : null,
                                     itemToUpdate.RecurringOrderProgramName)
-                                    .then(cart => {
-                                        if (cart) {
-                                            this.Cart = cart;
-                                        }
-                                    })
-                                    .finally(() => {
-                                        this.Mode.Loading = false;
-                                    });
                             }, 400);
                         }
 
                         this.debounceUpdateItem(id);
                     },
-
                     removeCartItem(index) {
                         var item = this.Cart.LineItemDetailViewModels[index];
-                        this.Mode.Loading = true;
+                        this.Steps.ReviewCart.Loading = true;
                         self.checkoutService.removeCartItem(item.Id, item.ProductId)
                             .then(cart => {
                                 if (cart) {
                                     this.Cart = cart;
                                 }
                             })
                             .finally(() => {
-                                this.Mode.Loading = false;
+                                this.Steps.ReviewCart.Loading = false;
                             });
 
                         this.Cart.LineItemDetailViewModels.splice(index, 1);
+                    },
+                    updateBeforeEditLineItemList() {
+                        this.beforeEditLineItemList = this.Cart.LineItemDetailViewModels.map(item => ({
+                            name: item.RecurringOrderFrequencyName,
```

Comment:
if this variable used for RecurringOrder - then move it to corresponding controller